### PR TITLE
GRASS: export temp data

### PR DESCRIPTION
**Pull request by pesekon2**
ArcGIS:

image

### CODE REVIEW

**Review by landam — CHANGES_REQUESTED** [review]
Currently GRASS produces:

```
tests/data/output/temp/
├── aoi_mask.tif
├── dem_filled.tif
├── dem_flowdir.tif
├── dem_slope_mask.tif
├── mat_boundary.asc
├── mat_boundary.asc.aux.xml
├── mat_boundary.prj
├── soilveg_aoi_b.tif
├── soilveg_aoi_k.tif
├── soilveg_aoi_n.tif
├── soilveg_aoi_pi.tif
├── soilveg_aoi_ppl.tif
├── soilveg_aoi_ret.tif
├── soilveg_aoi_s.tif
├── soilveg_aoi_tau.tif
├── soilveg_aoi_v.tif
├── soilveg_aoi_x.tif
├── soilveg_aoi_y.tif
├── stream.csv
└── stream_seg.tif
```

Still a lot of data layers compared to ArcGIS are missing.

File: tests/test_utils.py [hunk 1]
```diff
@@ -195,6 +195,7 @@ def __init__(self, runner, params_fn=None):
                 "table_soil_vegetation_fieldname": "soilveg",
                 "streams_channel_type_fieldname": "channel_id",
                 "output": self._output_dir,
+                't': True
```
Comment (landam) [inline tests/test_utils.py:198]: @pesekon2 What is a purpose of running tests with `-t` flag?

File: bin/base/__init__.py [hunk 2]
```diff
@@ -63,5 +64,8 @@ def __init__(self, label, section_arguments=()):
         'Computation options',
         ('output', 'max_time_step', 'total_time')
     ),
-    Section('Advanced', ())  # TODO: Add ('preparation_only',))
+    Section(
```
Comment (landam) [inline bin/base/__init__.py:67]: QGIS should also add exported temp layers into map canvas.

File: smoderp2d/providers/grass/data_preparation.py [hunk 3]
```diff
@@ -685,3 +666,51 @@ def _get_field_names(self, ds):
...
+                kwargs['output']
+                # extension will be added specifically in export (raster/vector)
+            )
+            module = args[0]
+            if module[
+                0] == 'r' and module != 'r.to.vect' or module == 'v.to.rast':
+                self._run_grass_module(
+                    'r.out.gdal',
+                    input=kwargs['output'],
+                    format='GTiff',
```
Comment (landam) [inline smoderp2d/providers/grass/data_preparation.py:705]: Let's export data in the same format as core/control output.

Comment (landam) [reply smoderp2d/providers/grass/data_preparation.py:705]:
```suggestion
                    format='AAIGrid',
                    nodata=GridGlobals.NoDataValue,
```

**Review by landam — CHANGES_REQUESTED** [review]

**Comment by landam** [comment]
Some changes introduced in 7971ebe:

- use pygrass module interface instead of kwargs
- use writter to export raster 
- use writer to get file output path

**Comment by landam** [comment]
Currently missing layers:

- [ ] dem_aspect
- [ ] dem_flowacc
- [ ] dem_slope

File: bin/base/__init__.py (hunk 2 above)
Comment (landam) [reply bin/base/__init__.py:67]: See https://github.com/storm-fsv-cvut/smoderp2d/issues/228#issuecomment-1762757992

Comment (pesekon2) [reply bin/base/__init__.py:67]: Done in [cc90cf1](https://github.com/storm-fsv-cvut/smoderp2d/pull/284/commits/cc90cf12c163dfab85b917ebb1059715b392cbaf). Thanks bro.

**Review by landam — CHANGES_REQUESTED** [review]
- `temp` subgroup collapsed and unchecked
- add `control` subgroup (collapsed and unchecked)
- add `control_points` subgroup (collapsed and unchecked)

**Review by landam — APPROVED** [review]

**Pull request merged**